src/libtest/lib.rs

```diff
@@ -519,7 +530,13 @@ pub fn parse_opts(args: &[String]) -> Option<OptRes> {
         None if quiet => OutputFormat::Terse,
         Some("pretty") | None => OutputFormat::Pretty,
         Some("terse") => OutputFormat::Terse,
-        Some("json") => OutputFormat::Json,
+        Some("json") => {
+            if !allow_unstable {
```

Comment:
This is not quite what @alexcrichton mentioned in https://github.com/rust-lang/rust/pull/46450#issuecomment-351460410: There should be a `-Z unstable-options` flag, which may only be passed on `nightly` (the `allow_unstable()` function above), and only if `-Z unstable-options` is passed should json output be allowed (in other words: the user needs to explicitly enable unstable flags, before being allowed to pass the json flag).

Reply:
Ohh, got it.